src/main/java/org/truffleruby/core/numeric/FixnumNodes.java

```diff
@@ -1329,13 +1332,20 @@ public long powTwoLong(int a, int b) {
         }, limit = "getLimit()")
         public Object powConstantExponent(Object base, int exponent,
                 @Cached("exponent") int cachedExponent,
+                @Cached("create()") BranchProfile overflowProfile,
                 @Cached("create()") IntegerMulNode mulNode) {
             Object result = 1;
             int exp = cachedExponent;
             while (exp > 0) {
                 if ((exp & 1) == 0) {
                     base = mulNode.executeMul(base, base);
                     exp >>= 1;
+
+                    if (base instanceof DynamicObject) {
+                        overflowProfile.enter();
```

Comment:
Why isn't this a condition profile?

Reply:
Because this is the bad branch, and the other branch is always hit.